war/src/main/java/io/lumeer/engine/controller/UserGroupFacade.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) 2016 - 2017 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.engine.controller;
+
+import io.lumeer.engine.annotation.SystemDataStorage;
+import io.lumeer.engine.api.LumeerConst;
+import io.lumeer.engine.api.LumeerConst.Group;
+import io.lumeer.engine.api.LumeerConst.UserGroup;
+import io.lumeer.engine.api.data.DataDocument;
+import io.lumeer.engine.api.data.DataFilter;
+import io.lumeer.engine.api.data.DataStorage;
+import io.lumeer.engine.api.data.DataStorageDialect;
+import io.lumeer.engine.api.exception.UserAlreadyExistsException;
+import io.lumeer.engine.util.ErrorMessageBuilder;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+/**
+ * Stores users and their groups
+ */
+@RequestScoped
```

Comment:
Why `@RequestScoped`? There is no need for this. It should be `@SessionScoped`. In the future, it can even be `@ApplicationScoped` if we find a way to make system data storage application-scoped bean.